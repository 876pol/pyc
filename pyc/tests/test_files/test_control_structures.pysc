/*
    This file contains code for automated testing of control structures in pyc.
*/

int main() {
    // for loops
    for (int i = 0; i < 5; i += 1) print((string) i);
    print("\n");

    for (int i = 0; i < 10; i += 2) {
        if (i == 6) break;
        print((string) i);
    }
    print("\n");

    for (int i = 0; i < 10; i += 1) {
        if (i % 2 == 0) continue;
        print((string) i);
    }
    print("\n");

    // while loops
    int i = 0;
    while (i < 5) {
        print((string) i);
        i += 1;
    }
    print("\n");

    i = 0;
    while (i < 10) {
        if (i == 6) break;
        print((string) i);
        i += 2;
    }
    print("\n");

    i = 0;
    while (i < 10) {
        if (i % 2 == 0) {
            i += 1;
            continue;
        }
        print((string) i);
        i += 1;
    }
    print("\n");

    i = 1;
    do {
        i *= -2;
        if (i == -8) continue;
        print((string) i);
    } while (i != 16);
    print("\n");

    // nested loops
    string s = "outside";
    for (int i = 0; i < 5; i += 1) {
        string s = "outer";
        for (int j = 0; j < i; j += 1) {
            string s = "";
            print((string) i + " " + (string) j + "\n");
            {
                s = "inner";
                print(s + "\n");
            }
        }
        print(s + "\n");
    }
    print(s + "\n");

    // if-else statements
    int a = -1;
    if (a < 2)
        print("a < 2\n");
    else if (a < 5)
        print("2 <= a < 5\n");
    else
        print("5 <= a\n");

    a = 3;
    if (a < 2) {
        print("a < 2\n");
    } else if (a < 5) {
        print("2 <= a < 5\n");
    } else {
        print("5 <= a\n");
    }
    a = 100;
    if (a < 2) {
        print("a < 2\n");
    } else if (a < 5) {
        print("2 <= a < 5\n");
    } else {
        print("5 <= a\n");
    }
    return 0;
}