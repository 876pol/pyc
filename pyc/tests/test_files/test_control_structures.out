01234
024
13579
01234
024
13579
-2416
outer
1 0
inner
outer
2 0
inner
2 1
inner
outer
3 0
inner
3 1
inner
3 2
inner
outer
4 0
inner
4 1
inner
4 2
inner
4 3
inner
outer
outside
a < 2
2 <= a < 5
5 <= a
